src/helpers/detectHostComponentNames.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { Text, TextInput, View } from 'react-native';
+import TestRenderer from 'react-test-renderer';
+import { HostComponentNames } from '../config';
+import { getQueriesForElement } from '../within';
+
+export function detectHostComponentNames():
```

Comment:
Wouldn't be actually simpler to throw if we do not find relevant names?

Reply:
I thought about it but I'm not a huge fan of throwing functions, it's not explicit that a function may throw and it may not be handled somewhere and I don't think we'd want the user to see this error so I like better returning null so that typescript forces to handle the case

Reply:
One way to show your function is throwing is with JS doc, but I agree it's not as visible as null typing

Reply:
I would insist here on either return `null` instead of `{ text: null; textInput: null }` or throw error to be caught in the calling function. The current solution with `null` fields seems strange.

Reply:
I'm now returning an error message or the host component names. I threw at first but it's more complicated to manipulate errors since they aren't typed and since I wanted to display the error of the query this seemed to be the best way